projects/ccd-case-ui-toolkit/src/lib/shared/components/case-editor/services/cases.service.ts

```diff
@@ -103,19 +103,26 @@ export class CasesService {
 
     const loadingToken = this.loadingService.register();
 
-    return this.http
-      .get(url, { headers, observe: 'body' })
-      .pipe(
-        catchError(error => {
-          console.error('Error while getting case view with getCaseViewV2!');
-          console.error(error);
-          this.errorService.setError(error);
-          return throwError(error);
-        }),
-        finalize(() => this.finalizeGetCaseViewWith(caseId, loadingToken))
-      );
+    let http$ = this.http.get(url, { headers, observe: 'body' });
+
+    this.retryUtil.pipeTimeoutMechanismOn(http$, this.appConfig.getEnvironment() === 'aat', this.appConfig.getCaseRetrievalTimeouts());
```

Comment:
Applying the retry mechanism to any sort of Observer (be it for another API call or for any other "observation")  takes only this 1 line.